### PR TITLE
[HTMLSelect] fix option labels display & use `value` as label if not specified

### PR DESCRIPTION
**Pull request by jaamison**
#### Changes proposed in this pull request:

This PR addresses 2 related but different issues with the core/HTMLSelect component:

**1) Options' labels are not being rendered**

I first noticed this on the Popover example page:
screen shot 2018-07-11 at 12 22 48 pm

Apparently, React (16.2 at least) does not respect use of the `label` HTML attribute in lieu of child text nodes for `<option>` elements (upstream bug?).

This PR reverts to specifying labels for `<option>` items as inner HTML, rather than specifying them via the element's `label` attribute, which appears to display correctly:
screen shot 2018-07-11 at 12 23 11 pm

**2) If consumer supplies an array of `{value, label}` plain objects as `options={...} `, the component does not fallback to using the value of `value` as the `label` in cases where `label` is either not supplied or is falsy.**

I understand that it might seem logical to put the burden of ensuring proper argument syntax on the consumer, but I can foresee cases where this fallback logic would avoid unpleasant surprises, even for consumers who are aware of the proper syntax.

For instance, if consumer supplies:

`options={ [{value:1, label:"one"}, {value:2}, {value:3, label:"three"}, {value:4, label:undefined}] }`

the the current behavior would be to render options 2 & 4 as empty. The logic in this PR would at least render "2" & "4" respectively.

### CODE REVIEW

File: packages/core/src/components/html-select/htmlSelect.tsx [hunk 1]
```diff
@@ -66,19 +66,34 @@ export class HTMLSelect extends React.PureComponent<IHTMLSelectProps> {
...
+                      };
+            return <option key={option.value} children={option.label} {...option} />;
+        });
+    }
+
+    private coercePlainObj(option: IHTMLOptionProps) {
+        return {
+            label: option.label || option.value.toString(),
+            value: option.value,
+        };
```
Comment (giladgray) [inline packages/core/src/components/html-select/htmlSelect.tsx]: this function is too similar to lines 85-88. DRY please.

**Comment by giladgray** [comment]
pushed my code above to #2681

**Comment by giladgray** [comment]
@jaamison thanks for pushing this up and pointing out these issues! i fixed the component myself in the interest of time -- ready to cut the patch release asap.

**Comment by jaamison** [comment]
awesome, much more concise

**Pull request closed without merging**